oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/LogQueryEsDAO.java

```diff
@@ -61,22 +73,30 @@ public Logs queryLogs(String metricName, int serviceId, int serviceInstanceId, S
         if (startSecondTB != 0 && endSecondTB != 0) {
             mustQueryList.add(QueryBuilders.rangeQuery(Record.TIME_BUCKET).gte(startSecondTB).lte(endSecondTB));
         }
-
-        if (serviceId != Const.NONE) {
+        if (isNotEmpty(serviceId)) {
             boolQueryBuilder.must().add(QueryBuilders.termQuery(AbstractLogRecord.SERVICE_ID, serviceId));
         }
-        if (serviceInstanceId != Const.NONE) {
+        if (isNotEmpty(serviceInstanceId)) {
             boolQueryBuilder.must()
                             .add(QueryBuilders.termQuery(AbstractLogRecord.SERVICE_INSTANCE_ID, serviceInstanceId));
         }
-        if (StringUtil.isNotEmpty(endpointId)) {
+        if (isNotEmpty(endpointId)) {
             boolQueryBuilder.must().add(QueryBuilders.termQuery(AbstractLogRecord.ENDPOINT_ID, endpointId));
         }
-        if (!Strings.isNullOrEmpty(stateCode)) {
-            boolQueryBuilder.must().add(QueryBuilders.termQuery(AbstractLogRecord.STATUS_CODE, stateCode));
+        if (isNotEmpty(endpointName)) {
+            String matchCName = MatchCNameBuilder.INSTANCE.build(AbstractLogRecord.ENDPOINT_NAME);
+            mustQueryList.add(QueryBuilders.matchPhraseQuery(matchCName, endpointName));
         }
-        if (!Strings.isNullOrEmpty(traceId)) {
-            boolQueryBuilder.must().add(QueryBuilders.termQuery(TRACE_ID, traceId));
+        if (nonNull(relatedTrace)) {
+            if (isNotEmpty(relatedTrace.getTraceId())) {
+                boolQueryBuilder.must()
+                                .add(QueryBuilders.termQuery(AbstractLogRecord.TRACE_ID, relatedTrace.getTraceId()));
+            }
+            if (isNotEmpty(relatedTrace.getSegmentId())) {
+                boolQueryBuilder.must().add(
+                    QueryBuilders.termQuery(AbstractLogRecord.TRACE_SEGMENT_ID, relatedTrace.getSegmentId()));
+            }
+            // TODO add span id
```

Comment:
Don't forget this. Seems existing in all `ILogQueryDAO` implementations.

Reply:
done